nbviewer/handlers.py

```diff
@@ -45,12 +47,25 @@ class CreateHandler(BaseHandler):
 
     only redirects to the appropriate URL
     """
+    uri_rewrite_list = None
+
     def post(self):
         value = self.get_argument('gistnorurl', '')
-        redirect_url = transform_ipynb_uri(value, self.provider_rewrites)
+        redirect_url = transform_ipynb_uri(value, self.get_provider_rewrites())
         app_log.info("create %s => %s", value, redirect_url)
         self.redirect(redirect_url)
 
+    @classmethod
+    def get_provider_rewrites(cls):
+        # storing this on a class attribute is a little icky, but is better
+        # than the global this was refactored from.
+        if cls.uri_rewrite_list is None:
+            # providers is a list of module import paths
+            providers = self.settings['provider_rewrites']
```

Comment:
Ah, a bit more involved than that, since `settings` actually would only be on `self`. hmm...